### PR TITLE
Simplify `EsHeaderComponent` implementation.

### PR DESCRIPTION
**Pull request by alexeykostevich**
- Update the component to use Ember Octane API.
- Remove the code for reseting `expanded` on CSS breakpoints for
  a more predictable behaviour and prevention of memory leaks.
- Add `navbar-expanded` class to simplify styles and reduce coupling
  with HTML structure.
- Fix accessibility warnings.
- Add additional tests.

### CODE REVIEW

**Comment by alexeykostevich** [comment]
Recreate a PR to address the issues with history in https://github.com/ember-learn/ember-styleguide/pull/303.

**Comment by mansona** [comment]
@alexeykostevich thanks for updating the PR 👍 it makes it much easier to review now 🎉

File: addon/components/es-header.js [hunk 1]
```diff
@@ -1,43 +1,24 @@
...
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
 
 import defaultLinks from '../constants/links';
 
+const defautHomePage = 'https://www.emberjs.com';
+
 export default class EsHeaderComponent extends Component {
-  expanded = false;
+  @tracked expanded = false;
```
Comment (mansona) [inline addon/components/es-header.js:10]: are you able to do this without tracked? and using `this.set()` 🤔 it's just that this isn't working in Ember 3.8 now (CI is failing) and I think using tracked is probably the issue.

Comment (alexeykostevich) [reply addon/components/es-header.js]: Done :)

**Comment by mansona** [comment]
@alexeykostevich thanks for doing this, and thanks for sticking with it 🎉

**Pull request merged**